### PR TITLE
Fix/relaying and downloading speeds

### PR DESCRIPTION
**Pull request by jcnelson**
Fixes relaying and makes block downloads go faster.  Also more debugging pleasentries.

### CODE REVIEW

File: src/net/p2p.rs [hunk 1]
```diff
@@ -1625,46 +1633,54 @@ impl PeerNetwork {
...
-                        // skip this step -- no DNS client available
-                        test_debug!("{:?}: no DNS client provided; skipping block download", &self.local_peer);
-                        self.work_state = PeerNetworkWorkState::Prune;
                     }
+                },
+                PeerNetworkWorkState::Prune => {
+                    // clear out neighbor connections after we finish sending
+                    if self.do_prune {
+                        do_prune = true;
+                        self.do_prune = false;
```
Comment (kantai) [inline src/net/p2p.rs:1673]: Will this function loop until self.do_prune is set to true by some call to `do_network_neighbor_walk` gets a `walk_result` that requests a prune?

Comment (jcnelson) [reply src/net/p2p.rs:1673]: ah, typo.  Meant to just make the loop break once the Prune state is reached

Comment (jcnelson) [reply src/net/p2p.rs:1673]: Lemme change that and re-test it.  I just wanted the loop to break once the network state did one round of work.

Comment (jcnelson) [reply src/net/p2p.rs:1677]: Addressed in e0349d2

**Comment by lgalabru** [comment]
Some of the tests failing in this PR are being fixed with https://github.com/blockstack/stacks-blockchain/pull/1483

**Pull request merged**